src/ProjectFileManager.cpp

```diff
@@ -1311,15 +1316,108 @@ class ImportProgress final
          }
       }
    }
-   
+
 private:
-   
+
    ImportFileHandle* mImportFileHandle {nullptr};
    std::unique_ptr<BasicUI::ProgressDialog> mProgressDialog;
 };
 
+namespace {
+void DoUseMirResultToConfigureProject(AudacityProject& project, double qpm)
+{
+   auto& projectTimeSignature = ProjectTimeSignature::Get(project);
+   projectTimeSignature.SetTempo(qpm);
+   // For now, our MIR only consists of filename parsing. If that succeeeds,
+   // most likely it is a 4/4 time signature. And if not, it still is something
+   // the user can adjust manually - until we get smarter.
+   projectTimeSignature.SetLowerTimeSignature(4);
+   projectTimeSignature.SetUpperTimeSignature(4);
+}
 
+bool UserWantsMirResultToConfigureProject(AudacityProject& project)
+{
+   const auto policy = UseMirResultToConfigureProject.Read();
+   if (policy == wxString("Ask"))
+   {
+      AudacityDontAskAgainMessageDialog m(
+         &GetProjectPanel(project), XO("Import"),
+         XO("Use detected music information to configure project?"));
```

Comment:
Shouldn't we display detected bpm value in that dialog, to make user well informed about the choice?

Btw, I get strange results when trying to import "Big Bass Hit" from MuseHub. It sets my project tempo to 542,373

Reply:
Yo, me too. I messed up a constant during clean-up, fix in 6c471cd.